src/stores/signupStore.js

```diff
@@ -0,0 +1,48 @@
+import axios from "axios";
+import { defineStore } from "pinia";
+import { reactive, toRefs } from "vue";
+
+const baseUrl = process.env.VUE_APP_BASE_URL;
+
+export const useSignupStore = defineStore("signupStore", () => {
+  const state = reactive({
+    roles: [],
+    loading: false,
+    error: null,
+    isUserSignedup: false,
+    user: null,
+  });
+
+  const getRoles = async () => {
+    try {
+      const res = await axios.get(`${baseUrl}role/list`);
```

Comment:
define base url in axios interceptor and append from there @aayushbatra02

Reply:
ok sir